packages/go_router/lib/src/delegate.dart

```diff
@@ -0,0 +1,95 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'builder.dart';
+import 'configuration.dart';
+import 'match.dart';
+import 'matching.dart';
+import 'typedefs.dart';
+
+/// GoRouter implementation of the RouterDelegate base class.
+class GoRouterDelegate extends RouterDelegate<RouteMatchList>
+    with PopNavigatorRouterDelegateMixin<RouteMatchList>, ChangeNotifier {
+  /// Constructor for GoRouter's implementation of the
+  /// RouterDelegate base class.
+  GoRouterDelegate({
+    required RouteConfiguration configuration,
+    required GoRouterBuilderWithNav builderWithNav,
+    required GoRouterPageBuilder? errorPageBuilder,
+    required GoRouterWidgetBuilder? errorBuilder,
+    required List<NavigatorObserver> observers,
+    required this.routerNeglect,
+    String? restorationScopeId,
+  }) : builder = RouteBuilder(
+          configuration: configuration,
+          builderWithNav: builderWithNav,
+          errorPageBuilder: errorPageBuilder,
+          errorBuilder: errorBuilder,
+          restorationScopeId: restorationScopeId,
+          observers: observers,
+        );
+
+  /// Builds the top-level Navigator given a configuration and location.
+  final RouteBuilder builder;
```

Comment:
(I ask because there's a member below with `@visibleForTesting`)

Reply:
Yes, good catch, in fact, how do you feel about adding this to all of its methods? For example, the intended use is `GoRouter.of(context).push()` rather than `GoRouter.of(context).routerDelegate.push()`.

Reply:
That sounds good to me 👍

Reply:
I'll do this as part of https://github.com/flutter/flutter/issues/107757